app/controllers/products_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class ProductsController < ApplicationController
+  def index
+    render :index, locals: { products: Product.all }
+  end
+
+  def show
+    render :show, locals: { product: Product.find(params[:id]) }
```

Comment:
you can use memoization - add a private method

```
def product
    @product ||= Product.find(params[:id]
end
```

then you can use `product` in controller :)